pmd-java/src/main/java/net/sourceforge/pmd/lang/java/rule/design/CyclomaticComplexityRule.java

```diff
@@ -139,11 +139,17 @@ public Object visit(ASTAnyTypeDeclaration node, Object data) {
 
 
     @Override
-    public final Object visit(ASTMethodOrConstructorDeclaration node, Object data) {
+    public final Object visit(MethodLikeNode node, Object data) {
 
         int cyclo = (int) JavaMetrics.get(JavaOperationMetricKey.CYCLO, node, cycloOptions);
         if (cyclo >= methodReportLevel) {
-            addViolation(data, node, new String[]{node instanceof ASTMethodDeclaration ? "method" : "constructor",
+            String nodeType = node.getKind() == MethodLikeKind.METHOD
+                    ? "method"
+                    : node.getKind() == MethodLikeKind.CONSTRUCTOR
+                            ? "constructor"
+                            : "lambda";
```

Comment:
sounds like a name for `MethodLikeKind`, something that should be part of the enum itself possibly

Reply:
I think I can open another PR for this small changeset